src/patroni.py

```diff
@@ -476,16 +477,16 @@ def last_postgresql_logs(self) -> str:
         if not container.can_connect():
             logger.debug("Cannot get last PostgreSQL log from Rock. Container inaccessible")
             return ""
-        log_files = container.list_files(POSTGRESQL_LOGS_PATH, pattern=POSTGRESQL_LOGS_PATTERN)
-        if len(log_files) == 0:
-            return ""
-        log_files.sort(key=lambda f: f.path, reverse=True)
         try:
+            log_files = container.list_files(POSTGRESQL_LOGS_PATH, pattern=POSTGRESQL_LOGS_PATTERN)
+            if len(log_files) == 0:
+                return ""
+            log_files.sort(key=lambda f: f.path, reverse=True)
             with container.pull(log_files[0].path) as last_log_file:
                 return last_log_file.read()
-        except OSError as e:
+        except Error:
```

Comment:
I don't think accessing the container will raise an `OSError` and list files will error if the directory doesn't exist.